packages/sdk/src/Fuel.ts

```diff
@@ -1,3 +1,4 @@
+import type { WalletProvider } from '@fuel-wallet/types';
```

Comment:
regarding provider name:
- it defines how the user will connect with the network. will the user connect to `fuel-beta-3` production node directly? or will use some other provider like `alchemy`?


for this feature I think a best name would be: `WalletConnector`

then we allow other apps to create their own `Connectors` extending from `WalletConnector`. and we also can expose our won `FuelWalletConnector`
![image](https://user-images.githubusercontent.com/8636507/235786716-85415c17-32c3-4d80-a58a-13e77212b993.png)

Reply:
adding to this idea, I think we should introduce other layer ahead of the connectors.

then injectFuel we can call the `FuelClient` which can receive options like:

```
new FuelClient({
  connectors: [
    new FuelWalletConnector(),
    new FueletWalletConnector(),
    new WatheverWalletConnector(),
  ]
});
```

then the `WalletConnectors` can have functions like:
- `isConnected` - is connected through this connector
- `isReady` - is Wallet available in user browser
- others..

on this way we can add to top layer `FuelClient` methods like:
`getConnectedWallet()` - will return `WalletConnector` that was used to connect
`getPossibleWallets()` - will return `WalletConnector`s that are ready to use in user browser

then we can remove the concept of `targetWallet`, because client will handle it

Reply:
This can make sense right now we don't have a good way to provide the SDK to work alone on the FE. We need to first close #382 to think about this approach.

We want to provide a more integrated experience where the developer don't need to install multiple things in order to get access to all wallets the users has installed.

Reply:
yeah providing a `FuelClient` is good solution for solving  #382 .

idea is that client can be created by just importing it, don't need to wait anything in `window`
```
import { FuelClient } from "@fuels-wallet/sdk";

new FuelClient({
  connectors: [
    new FuelWalletConnector(),
    new FueletWalletConnector(),
    new WatheverWalletConnector(),
  ]
});
```

then the client itself will search for window dependencies like `window.fuel` and other possibles.

this also allow us to in the future to create a react specific `FuelProvider` informing the `client`, and build our own experience of connecting wallet, with a cool connect popup and great ux. also to provide specific `hooks` to interact with wallets